### PR TITLE
CBL-3827: TLS by default in CollectionSGTest

### PR DESCRIPTION
**Pull request by callumbirks**
Make TLS default in CollectionSGTest. 
Load cert in CollectionSGTest constructor.
Update C4Test::readFile() to modern C++. 
Add findProjectRoot() function to help with finding paths.
All tests passing locally.

### CODE REVIEW

**Comment by callumbirks** [comment]
This does accidentally include a commit from another branch as I checked out to this branch from that one, but I have removed those changes from this branch.

**Comment by callumbirks** [comment]
Need to get this merged ASAP as the TLS config in SGW is causing issues with running some tests.
Have tested with these changes and LiteCore and CollectionSGTests are all passing.

**Review by velicuvlad — APPROVED** [review]

**Comment by callumbirks** [comment]
The changes in LiteCore/tests/CMakeLists.txt copy the new certs I made (that are also used by SGW config) into the test executable dir so they can be read by the tests.

File: Replicator/tests/ReplicatorCollectionSGTest.cc [hunk 1]
```diff
@@ -111,6 +120,8 @@ class ReplicatorCollectionSGTest : public ReplicatorAPITest {
         }
     }
 
+
+
```
Comment (pasin) [inline Replicator/tests/ReplicatorCollectionSGTest.cc:124]: Some extra lines here.

File: C/tests/c4Test.cc [hunk 2]
```diff
@@ -627,23 +627,16 @@ void C4Test::checkAttachments(C4Database *inDB, vector<C4BlobKey> blobKeys, vect
...
-    fleece::alloc_slice result(size);
-    ssize_t bytesRead = fread((void*)result.buf, 1, size, fd);
-    REQUIRE(bytesRead == size);
-    fclose(fd);
+// Parameter is relative filepath for cert from project root
+fleece::alloc_slice C4Test::readFile(std::string filepath) {
+    std::ifstream inFile(filepath);
+    REQUIRE(inFile.is_open());
+    std::stringstream outData;
+    outData << inFile.rdbuf();
```
Comment (jianminzhao) [inline C/tests/c4Test.cc:635]: Check io read by size, as something like, " REQUIRE(bytesRead == size); " in the above.

**Review by pasin — APPROVED** [review]

**Pull request merged**